modules/mono/editor/bindings_generator.cpp

```diff
@@ -1418,16 +1421,38 @@ Error BindingsGenerator::_generate_cs_type(const TypeInterface &itype, const Str
 	output.append(itype.proxy_name);
 
 	if (is_derived_type && !itype.is_singleton) {
+		property_names.append(INDENT1 "public new class PropertyName");
+		method_names.append(INDENT1 "public new class MethodName");
+		signal_names.append(INDENT1 "public new class SignalName");
```

Comment:
The classes should be static. I don't see why they need to inherit the base one when the fields are static.

Reply:
not inherit, can only CanvasItem.PropertyName.Visible;
inherit, can Node2D.PropertyName.Visible;

Reply:
I am trying to migrate my 3. X project to 4.0.
I will study the code of 4.0 and see how to change this PR.

Reply:
Oh, if C# can resolve static fields from base classes that way, then this makes sense 👍